modules/sor/sorV2/lib/pathGraph/pathGraph.ts

```diff
@@ -457,38 +506,75 @@ export class PathGraph {
         return filtered;
     }
 
-    private getLimitAmountSwapForPath(path: PathGraphEdgeData[], swapKind: SwapKind): bigint {
-        let limit = path[path.length - 1].pool.getLimitAmountSwap(
-            path[path.length - 1].tokenIn,
-            path[path.length - 1].tokenOut,
-            swapKind,
-        );
-
-        for (let i = path.length - 2; i >= 0; i--) {
-            const poolLimitExactIn = path[i].pool.getLimitAmountSwap(
-                path[i].tokenIn,
-                path[i].tokenOut,
+    private getLimitAmountSwapForPath(path: PathGraphEdgeData[]): bigint {
```

Comment:
I'm confused with this `getLimitAmountSwapForPath` method.
Even the previous implementation seems to be comparing amountsIn with amountsOut, which doesn't make sense to me 😕 
Do you mind explaining how it works, so I can understand the changes you made?

Reply:
No, of course not, this method finds the limit amount of the input/output token for the path, this limit is marked in a way that no token/bpt balance of any pool of the path will be changed more than 30%